.github/workflows/ci_cd.yaml

```diff
@@ -1,116 +1,155 @@
-name: Tests
+name: Tests and code quality
 
-on:
-  pull_request:
-  push:
-    branches:
-      - main
+on: [push, pull_request]
 
 jobs:
-  build-linux:
+  push_job:
+    if: ${{ github.event_name == 'push' }}
     runs-on: ubuntu-latest
-
     steps:
     - name: Checkout Repository
-      uses: actions/checkout@v2
-
-    - name: Set up Python
-      uses: actions/setup-python@v2
-      with:
-        python-version: '3.9'
-
+      uses: actions/checkout@v4
     - name: Install and configure Poetry
       uses: snok/install-poetry@v1
       with:
         version: 1.5.1
-        virtualenvs-create: false
-
+        virtualenvs-create: true
```

Comment:
@nilskre why do we switch to using venvs?

Reply:
Not particularly, just to keep it simple. These are good reasons, I just wanted to know why. :)